File: app/src/main/java/cat/xojan/random1/injection/component/HomeComponent.java
package cat.xojan.random1.injection.component;

import cat.xojan.random1.injection.PerActivity;
import cat.xojan.random1.injection.module.BaseActivityModule;
import cat.xojan.random1.injection.module.HomeModule;
import cat.xojan.random1.ui.activity.HomeActivity;
import cat.xojan.random1.ui.fragment.DownloadsFragment;
import cat.xojan.random1.ui.fragment.PodcastListFragment;
import cat.xojan.random1.ui.fragment.ProgramFragment;
import dagger.Component;

@PerActivity
@Component(
        dependencies = AppComponent.class,
        modules = {
                BaseActivityModule.class,
                HomeModule.class
        }
)
public interface HomeComponent extends BaseActivityComponent {
    void inject(HomeActivity homeActivity);
    void inject(PodcastListFragment podcastListFragment);
    void inject(ProgramFragment programFragment);
    void inject(DownloadsFragment downloadedPodcastFragment);
}

File: app/src/main/java/cat/xojan/random1/injection/component/HomeComponent.kt
package cat.xojan.random1.injection.component

import cat.xojan.random1.feature.home.DownloadsFragment
import cat.xojan.random1.feature.home.HomeActivity
import cat.xojan.random1.feature.home.ProgramFragment
import cat.xojan.random1.injection.PerActivity
import cat.xojan.random1.injection.module.BaseActivityModule
import cat.xojan.random1.injection.module.HomeModule
import dagger.Component

@PerActivity
@Component(
        dependencies = [(AppComponent::class)],
        modules = [(BaseActivityModule::class), (HomeModule::class)])
interface HomeComponent : BaseActivityComponent {
    fun inject(homeActivity: HomeActivity)
    fun inject(programFragment: ProgramFragment)
    fun inject(downloadedPodcastFragment: DownloadsFragment)
}